### ISSUE TITLE
OSX Client: Modifier Keys don't work in remote control applications

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#204]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/204
> 
> Issue created on: 2018-12-19 by @githubjsorg
> Issue last updated on: 2019-01-05

### Operating Systems ###

Server: Windows Server 2008

Client: macOS 10.12.6 Sierra

### Barrier Version ###

2.1.0

### Steps to reproduce bug ###

1. Connect to a remote machine with one of these (MS Remote Desktop, NoMachine)
2. Try to use SHIFT, CTRL, ALT, SUPER.
3. These keys are not recognized as being depressed on the remote machine.

### Other info ###

* When did the problem start to occur? When I switched to Barrier. I used to use Synergy 1.3.
* Is there a way to work around it? No
* Does this bug prevent you from using Barrier entirely? No

Interestingly CTRL, ALT, and SHIFT work just fine in VNC but not SUPER.

I only switched to Barrier after updating from Maverics to Sierra because Synergy wouldn't recognize modifier keys at all on the client. Using Synergy in Maverics, modifier keys worked just fine even in remote control apps.

---

> Commented on: 2018-12-21 by @AdrianKoshka

Hey, thanks for submitting an issue. I'll get back to this probably after December 26th. Visiting relatives for the holidays right now.

---

> Commented on: 2018-12-21 by @githubjsorg

You're welcome. Have fun with your relatives. :)

When you get back, let me know if I can help with any additional information or testing fixes.

---

> Commented on: 2019-01-02 by @githubjsorg

Adrian, thank you for the update. How do I get a new binary(s) to test with? And do I need a new one for both win and macos or just 1 or the other?

---

> Commented on: 2019-01-02 by @AdrianKoshka

You would have to build barrier yourself from source for both platforms from master, I have no ETA on when 2.2 will be released.

---

> Commented on: 2019-01-04 by @githubjsorg

I see a guide on how to build on Windows, which I might be able to do.
https://github.com/debauchee/barrier/wiki/Building-on-Windows
Any chance this will work with VS 2017?

But I don't see a guide on how to build on macos.

---

> Commented on: 2019-01-04 by @AdrianKoshka

> Any chance this will work with VS 2017?

Don't know, I just use Visual Studio Community because the instructions say to

> But I don't see a guide on how to build on macos.

No one has written one, sorry.

---

> Commented on: 2019-01-05 by @p12tic

@githubjsorg See what we do in .travis.yml.

Basically, on OSX you need to install ports and do the following:

```
sudo port install qt5-qtbase openssl
sh -x ./clean_build.sh
```

---

> Commented on: 2019-01-05 by @AdrianKoshka

@p12tic Do you have access to the wiki? If so, writing up a "how to compile" page for macOS would be helpful if you know how.

---

> Commented on: 2019-01-05 by @p12tic

I don't have access unfortunately.

---

> Commented on: 2019-01-05 by @AdrianKoshka

Darn. I mean, you could write up something on https://gist.github.com/ and then when you're happy with it, link me to it so I can fold it into the wiki, if you want that is.

---

> Closed on: 2018-12-30 by @AdrianKoshka
